drivers/dma/dma_stm32.c

```diff
@@ -0,0 +1,599 @@
+{
+	u32_t periph, memory;
+	u32_t m_size = 0, p_size = 0;
+
+	if (source_periph) {
+		periph = config->source_data_size;
+		memory = config->dest_data_size;
+	} else {
+		periph = config->dest_data_size;
+		memory = config->source_data_size;
+	}
+	int index = find_lsb_set(config->source_data_size) - 1;
+
+	m_size = table_m_size[index];
+	index = find_lsb_set(config->dest_data_size) - 1;
+	p_size = table_p_size[index];
+
+#if defined(CONFIG_DMA_STM32_V1)
+	int ret = stm32_dma_burst_config(config, source_periph, dma, id);
+
+	if (ret < 0) {
+		return ret;
+	}
+#endif
+
+	return (m_size | p_size);
+}
+
+static int dma_stm32_get_priority(u8_t priority, u32_t *ll_priority)
+{
+	switch (priority) {
+	case STM32_DMA_PRIORITY_LOW:
+		*ll_priority = LL_DMA_PRIORITY_LOW;
+		break;
+	case STM32_DMA_PRIORITY_MEDIUM:
+		*ll_priority = LL_DMA_PRIORITY_MEDIUM;
+		break;
+	case STM32_DMA_PRIORITY_HIGH:
+		*ll_priority = LL_DMA_PRIORITY_HIGH;
+		break;
+	case STM32_DMA_PRIORITY_VERYHIGH:
+		*ll_priority = LL_DMA_PRIORITY_VERYHIGH;
+		break;
+	default:
+		LOG_ERR("Priority error. %d", priority);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_stm32_get_direction(enum dma_channel_direction direction,
+				   u32_t *ll_direction)
+{
+	switch (direction) {
+	case MEMORY_TO_MEMORY:
+		*ll_direction = LL_DMA_DIRECTION_MEMORY_TO_MEMORY;
+		break;
+	case MEMORY_TO_PERIPHERAL:
+		*ll_direction = LL_DMA_DIRECTION_MEMORY_TO_PERIPH;
+		break;
+	case PERIPHERAL_TO_MEMORY:
+		*ll_direction = LL_DMA_DIRECTION_PERIPH_TO_MEMORY;
+		break;
+	default:
+		LOG_ERR("Direction error. %d", direction);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_stm32_get_memory_increment(enum dma_addr_adj increment,
+					  u32_t *ll_increment)
+{
+	switch (increment) {
+	case DMA_ADDR_ADJ_INCREMENT:
+		*ll_increment = LL_DMA_MEMORY_INCREMENT;
+		break;
+	case DMA_ADDR_ADJ_NO_CHANGE:
+		*ll_increment = LL_DMA_MEMORY_NOINCREMENT;
+		break;
+	case DMA_ADDR_ADJ_DECREMENT:
+		return -ENOTSUP;
+	default:
+		LOG_ERR("Memory increment error. %d", increment);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_stm32_get_periph_increment(enum dma_addr_adj increment,
+					  u32_t *ll_increment)
+{
+	switch (increment) {
+	case DMA_ADDR_ADJ_INCREMENT:
+		*ll_increment = LL_DMA_PERIPH_INCREMENT;
+		break;
+	case DMA_ADDR_ADJ_NO_CHANGE:
+		*ll_increment = LL_DMA_PERIPH_NOINCREMENT;
+		break;
+	case DMA_ADDR_ADJ_DECREMENT:
+		return -ENOTSUP;
+	default:
+		LOG_ERR("Periph increment error. %d", increment);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_stm32_config(struct device *dev, u32_t id,
+			    struct dma_config *config)
+{
+	struct dma_stm32_data *data = dev->driver_data;
+	struct dma_stm32_stream *stream = &data->streams[id];
+	const struct dma_stm32_config *dma_stm32_config =
+					dev->config->config_info;
+	DMA_TypeDef *dma = (DMA_TypeDef *)dma_stm32_config->base;
+	u32_t p_addr, m_addr;
+	u16_t count;
+	int ret;
+
+	if (id >= data->max_streams) {
```

Comment:
For my information, for this function, have you considered use of `LL_DMA_Init` ?
If yes, what were the reasons why you chose not to use it?

Reply:
No, didn't know this before. I can use LL_DMA_Init instead, it seems like lines of code they will use are almost the same.